frontend/src/metabase/modes/lib/modes.ts

```diff
@@ -0,0 +1,49 @@
+import SegmentMode from "../components/modes/SegmentMode";
+import MetricMode from "../components/modes/MetricMode";
+import TimeseriesMode from "../components/modes/TimeseriesMode";
+import GeoMode from "../components/modes/GeoMode";
+import PivotMode from "../components/modes/PivotMode";
+import NativeMode from "../components/modes/NativeMode";
+import DefaultMode from "../components/modes/DefaultMode";
+import { QueryMode } from "metabase-types/types/Visualization";
```

Comment:
Deprecated `metabase-types/types` usage. Please use types from `metabase-types/api`. If they don't exist yet, please move this type there.

Reply:
I had a look at moving the necessary `types` to `api` but it seems to be a very big job as there is a lots of dependencies and out of the scope regarding that issue.